### PR TITLE
Make it build with ghc-9.10

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -11,8 +11,15 @@ repository cardano-haskell-packages
...
-  , cardano-haskell-packages 2023-05-16T03:39:10Z
+    hackage.haskell.org 2024-06-27T21:37:22Z
+  , cardano-haskell-packages 2024-06-27T10:53:24Z
+
+allow-newer:
+  , io-classes:base
+  , io-classes:bytestring
+  , serdoc-core:bytestring
+  , serdoc-core:template-haskell
+  , si-timers:base
```
Comment (erikd) [inline cabal.project:22]: Do we want to remove these before this is merged?

Comment (erikd) [reply cabal.project]: All these have been removed.

**Comment by erikd** [comment]
The build failure with `ghc-8.10` is due to toolchain issues.

File: typed-protocols-cborg/typed-protocols-cborg.cabal [hunk 2]
```diff
@@ -18,7 +18,7 @@ extra-source-files:  CHANGELOG.md, README.md
 library
   exposed-modules:   Network.TypedProtocol.Codec.CBOR
 
-  build-depends:     base            >=4.12  && <4.20,
+  build-depends:     base            >=4.12  && <5,
```
Comment (coot) [inline typed-protocols-cborg/typed-protocols-cborg.cabal:21]: Are you sure there won't be `4.21` in the future?

Comment (erikd) [reply typed-protocols-cborg/typed-protocols-cborg.cabal:21]: Current base is `4.21`, 

This is the same `base` bound as is already used in `typed-protocols-doc.cabal`

Comment (coot) [reply typed-protocols-cborg/typed-protocols-cborg.cabal:21]: I just overlooked it.  We should use `< 4.21`.

Comment (erikd) [reply typed-protocols-cborg/typed-protocols-cborg.cabal]: Updated.

**Review by coot — APPROVED** [review]

**Comment by coot** [comment]
To fix `8.10 macos-latest` we need to add something like this to the `.github/workflows/haskell.yml`: https://github.com/input-output-hk/io-sim/pull/153/commits/8484627a0f8d8252766f3ce56701cccbe12bc3be

**Comment by erikd** [comment]
> To fix `8.10 macos-latest` we need to add something like this to the `.github/workflows/haskell.yml`: [input-output-hk/io-sim@8484627](https://github.com/input-output-hk/io-sim/commit/8484627a0f8d8252766f3ce56701cccbe12bc3be)

Possibly, but add what?

**Comment by coot** [comment]
I pushed a commit which should fix it.

**Pull request merged**